MapboxVisionAR/ARRenderer.swift

```diff
@@ -439,15 +443,15 @@ class ARRenderer: NSObject, MTKViewDelegate {
         guard let commandEncoder = commandBuffer.makeRenderCommandEncoder(descriptor: renderPass)
         else { return }
 
-        if let frame = dataProvider.getCurrentFrame(), let texture = makeTexture(from: frame) {
+        if let frame = frame, let texture = makeTexture(from: frame) {
             commandEncoder.setRenderPipelineState(renderPipelineBackground)
             commandEncoder.setVertexBuffer(backgroundVertexBuffer, offset: 0, index: 0)
             commandEncoder.setFragmentTexture(texture, index: 0)
             commandEncoder.drawPrimitives(type: .triangle, vertexStart: 0, vertexCount: textureMappingVertices.count)
         }
-        
-        if let routeData = dataProvider.getARRouteData() {
-            drawScene(commandEncoder: commandEncoder, routeData: routeData)
+
+        if let lane = lane {
```

Comment:
Maybe we should not draw `Lane` instead of whole scene? It will allow to add new features in future easily.